system_metrics_collector/src/system_metrics_collector/cpu_data.hpp

```diff
@@ -0,0 +1,78 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SYSTEM_METRICS_COLLECTOR__CPU_DATA_HPP_
+#define SYSTEM_METRICS_COLLECTOR__CPU_DATA_HPP_
+
+#include <string>
+#include <sstream>
+
+/**
+ * Struct containing data parsed from a /proc/stat cpu line
+ */
+struct ProcCpuData
+{
+  static constexpr const char EMPTY_LABEL[] = "empty";
+  /**
+   * Enum representing each item in a /proc/stat cpu line
+   */
+  enum ProcCpuStates
```

Comment:
```suggestion
  enum class ProcCpuStates
```

Prefer enum class whenever possible (enum is older variant inherited from C)

https://www.geeksforgeeks.org/enum-classes-in-c-and-their-advantage-over-enum-datatype/

Reply:
Also, consider moving that at the top-level, using an `enum class` prevents naming clash.